packages/scenes/src/components/layout/grid/SceneGridLayout.tsx

```diff
@@ -355,8 +364,43 @@ export class SceneGridLayout extends SceneObjectBase<SceneGridLayoutState> imple
 
   public buildGridLayout(width: number, height: number): ReactGridLayout.Layout[] {
     let cells: ReactGridLayout.Layout[] = [];
+    const searchString = this.state.searchString;
+    const panelsPerRow = this.state.panelsPerRow;
 
-    for (const child of this.state.children) {
+    let children = this.state.children;
+    if ((searchString && searchString !== '') || (panelsPerRow && Number.isInteger(panelsPerRow))) {
+      if (this.savedLayout === undefined) {
+        this.savedLayout = this.state.children.map((v) => v.clone());
+      }
+
+      const panelFilterInterpolated = interpolate(this, searchString).toLowerCase();
+      const filteredChildren = this.state.children.filter(
+        (v) =>
+          typeof (v.state as any).body?.state?.title === 'string' &&
+          (v.state as any).body.state.title.toLowerCase().includes(panelFilterInterpolated)
+      );
```

Comment:
We can also interpolate the `title` , we could match variables that are in the title, something like: 

```suggestion
      const filteredChildren = this.state.children.filter(
        (v) => {
          const panelTitleInterpolated = interpolate(v, (v.state as any).body?.state?.title).toLowerCase();
          return typeof (v.state as any).body?.state?.title === 'string' &&
            (panelTitleInterpolated.includes(panelFilterInterpolated));
        }
      );

```